### PR TITLE
[Feature Fix]  [Firefox] Fix the incorrect time shown on file version table 

### LINKED ISSUE #2972: [Production & Staging] [Firefox] File version table has incorrect time
## Steps
1. Log in to osf.io on the Firefox browser
2. Go to a project you have admin on 
3. On the project overview page click on the osf storage folder to select it
4. In the toolbar, click to upload a file
5. Refresh the page and look at the project log.  You should see a log entry for the file you just uploaded.  Make note of the time showing.
6. Click on the link in the log to go to the file
7. Look to the right of the rendered image at the version table for the file
8. Make note of the time.

## Expected
I would expect the time showing in the project log for a file upload and the time in the version table for the file to be the same.  I would expect the time in the version table to be the accurate time.

## Actual
The log for the file upload is showing the correct time.  However, the version table shows the incorrect time; it is 4 hours ahead of the current time.

screenshot 2015-06-04 13 55 40
screenshot 2015-06-04 13 55 23

Appears to only happen on Firefox.

### PR DESCRIPTION
**Pull request by haoyuchen1992**
## Purpose
This PR is to solve the issue that on firefox it shows the wrong file modified date. 

Resolved #2972 

## Change
Different from all the other browsers, Firefox assumes  the input(`DateString`) of  `new Date(DateString)` is the date based on local time zone while all the other browsers treat the input based on UTC time zone. In our case, the input date string is UTC time-zone. Therefore, we need manually add offset  when displaying on firefox, by adding ''Z' at the end of date string. The letter Z ("Zulu") indicates  Coordinated Universal Time (see UTC±00:00).

So in summary, the PR code is to add 'Z' at the end of date time string. The action is only limited to Firefox. Since it changes the osfHelper.js, the solution will apply all the hidden date-shown problems on Firefox.

Reference material --[Stackoverflow Question](http://stackoverflow.com/questions/15109894/new-date-works-differently-in-chrome-and-firefox)

## Side effect
None.

### CODE REVIEW

**Comment by sloria** [comment]
Thanks :octopus:
